checks/evaluation/webhooks.go

```diff
@@ -20,42 +20,50 @@ import (
 	"github.com/ossf/scorecard/v4/checker"
 	sce "github.com/ossf/scorecard/v4/errors"
 	"github.com/ossf/scorecard/v4/finding"
+	"github.com/ossf/scorecard/v4/probes/webhooksWithoutSecret"
 )
 
 // Webhooks applies the score policy for the Webhooks check.
-func Webhooks(name string, dl checker.DetailLogger,
-	r *checker.WebhooksData,
+func Webhooks(name string,
+	findings []finding.Finding, dl checker.DetailLogger,
 ) checker.CheckResult {
-	if r == nil {
-		e := sce.WithMessage(sce.ErrScorecardInternal, "empty raw data")
+	expectedProbes := []string{
+		webhooksWithoutSecret.Probe,
+	}
+
+	if !finding.UniqueProbesEqual(findings, expectedProbes) {
+		e := sce.WithMessage(sce.ErrScorecardInternal, "invalid probe results")
 		return checker.CreateRuntimeErrorResult(name, e)
 	}
 
-	if len(r.Webhooks) < 1 {
-		return checker.CreateMaxScoreResult(name, "no webhooks defined")
+	if len(findings) == 1 && findings[0].Outcome == finding.OutcomeNotApplicable {
+		return checker.CreateMaxScoreResult(name, "project does not have webhook")
 	}
 
-	hasNoSecretCount := 0
-	for _, hook := range r.Webhooks {
-		if !hook.UsesAuthSecret {
-			dl.Warn(&checker.LogMessage{
-				Path: hook.Path,
-				Type: finding.FileTypeURL,
-				Text: "Webhook with no secret configured",
-			})
-			hasNoSecretCount++
+	var webhooksWithNoSecret int
+
+	totalWebhooks := findings[0].Values["totalWebhooks"]
```

Comment:
if we're doing 1 finding per webhook, do we need this `totalWebhooks` value still?

Reply:
Currently, we are doing one `OutcomeNegative` for each webhook without a secret. We need the total ammount of webhooks in the evaluation to calculate the proportion. The `totalWebhooks` value is the number of all webhooks in the project.

Reply:
we also create an `OutcomePositive` for webhooks which do have secrets. So wouldn't total webhooks be the number of findings?

Reply:
You're correct. Removed in https://github.com/ossf/scorecard/pull/3522/commits/2dc1b1e6f4d487dcd6afe334a5f533eae9a4fe09